src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -21,18 +21,28 @@ class ResourceService(val accessPolicyDAO: AccessPolicyDAO, val directoryDAO: Di
   }
 
   def listUserResourceActions(resourceType: ResourceType, resourceName: ResourceName, userInfo: UserInfo): Future[Set[ResourceAction]] = {
+    listResourceAccessPoliciesForUser(resourceType, resourceName, userInfo).map { matchingPolicies =>
+      matchingPolicies.flatMap(_.actions)
+    }
+  }
+
+  def listUserResourceRoles(resourceType: ResourceType, resourceName: ResourceName, userInfo: UserInfo): Future[Set[ResourceRoleName]] = {
+    listResourceAccessPoliciesForUser(resourceType, resourceName, userInfo).map { matchingPolicies =>
+      matchingPolicies.flatMap(_.role)
+    }
+  }
+
+  private def listResourceAccessPoliciesForUser(resourceType: ResourceType, resourceName: ResourceName, userInfo: UserInfo): Future[Set[AccessPolicy]] = {
```

Comment:
This only needs `name` from `resourceType` so I propose that it take a `ResourceTypeName` instead of a full `ResourceType`.